### PR TITLE
Notify on comments moderation

### PR DESCRIPTION
**Pull request by eikichi18**
Hi,

I've added notification about the publishing or elimination of a comment.

### CODE REVIEW

File: plone/app/discussion/browser/javascripts/moderation.js [hunk 1]
```diff
@@ -24,10 +24,10 @@ require([  // jshint ignore:line
     //#JSCOVERAGE_IF 0
 
     /**************************************************************************
-     * Window Load Function: Executes when complete page is fully loaded,
+     * Document Ready Function: Executes when complete page is fully loaded,
      * including all frames,
      **************************************************************************/
-    $(window).load(function () {
+    $(document).ready(function () {
```
Comment (ale-rt) [inline plone/app/discussion/browser/javascripts/moderation.js]: Isn't `$(document).ready` fired before `$(window).load`? If this is what you want I am fine with that. Comment might be updated.

Comment (eikichi18) [reply plone/app/discussion/browser/javascripts/moderation.js]: I'm agree with you $(document).ready is the first, but $(window).load it doesn't always work especially on FF and IE. Instead $(document).ready it seems to work properly. 

But maybe it’s better move this modify to another PR.

Comment (ale-rt) [reply plone/app/discussion/browser/javascripts/moderation.js]: As you prefer :) I'm not so much into javascript!

File: plone/app/discussion/browser/controlpanel.py [hunk 2]
```diff
@@ -17,6 +16,11 @@
 from zope.component import queryUtility
 from zope.component.hooks import getSite
 
+try:
+    from Products.CMFPlone.interfaces.controlpanel import IConfigurationChangedEvent  # noqa: E501
+except ImportError:
+    from Products.CMFPlone.interfaces import IConfigurationChangedEvent  # noqa: E501
```
[the remark this answers is not in the document]
Comment (eikichi18) [reply plone/app/discussion/browser/controlpanel.py]: Right! 
I'll explain the compatibility issue in a comment.

File: plone/app/discussion/events.py [hunk 3]
```diff
@@ -48,3 +50,15 @@ class ReplyAddedEvent(DiscussionEvent):
 class ReplyRemovedEvent(DiscussionEvent):
     """ Event to be triggered when a Comment reply is removed
     """
+
+
+@implementer(INotifyOnDelete)
+class NotifyOnDelete(DiscussionEvent):
+    """ Event to be triggered when a Comment is deleted
```
[the remark this answers is not in the document]
Comment (eikichi18) [reply plone/app/discussion/events.py]: I'll fix it.

File: plone/app/discussion/events.py [hunk 4]
```diff
@@ -48,3 +50,15 @@ class ReplyAddedEvent(DiscussionEvent):
...
+    """
+
+
+@implementer(INotifyOnPublish)
+class NotifyOnPublish(DiscussionEvent):
```
[the remark this answers is not in the document]
Comment (eikichi18) [reply plone/app/discussion/events.py]: I'll fix it.

File: plone/app/discussion/browser/javascripts/moderation.js (hunk 1 above)
Comment (eikichi18) [reply plone/app/discussion/browser/javascripts/moderation.js]: done :)

**Pull request merged**